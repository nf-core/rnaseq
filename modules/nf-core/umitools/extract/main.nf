process UMITOOLS_EXTRACT {
    tag "$meta.id"
    label "process_single"
    label "process_long"

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/umi_tools:1.1.5--py39hf95cd2a_0' :
        'biocontainers/umi_tools:1.1.5--py39hf95cd2a_0' }"

    input:
    tuple val(meta), path(reads)

    output:
    tuple val(meta), path("*.fastq.gz"), emit: reads
    tuple val(meta), path("*.log")     , emit: log
    path  "versions.yml"               , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    if (meta.single_end) {
        """
        umi_tools \\
            extract \\
            -I $reads \\
            -S ${prefix}.umi_extract.fastq.gz \\
            $args \\
            > ${prefix}.umi_extract.log

        cat <<-END_VERSIONS > versions.yml
        "${task.process}":
            umitools: \$( umi_tools --version | sed '/version:/!d; s/.*: //' )
        END_VERSIONS
        """
    }  else {
        """
        umi_tools \\
            extract \\
            -I ${reads[0]} \\
            --read2-in=${reads[1]} \\
            -S ${prefix}.umi_extract_1.fastq.gz \\
            --read2-out=${prefix}.umi_extract_2.fastq.gz \\
            $args \\
            > ${prefix}.umi_extract.log

        cat <<-END_VERSIONS > versions.yml
        "${task.process}":
            umitools: \$( umi_tools --version | sed '/version:/!d; s/.*: //' )
        END_VERSIONS
        """
    }
}
