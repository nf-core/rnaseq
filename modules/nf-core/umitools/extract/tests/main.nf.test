nextflow_process {

    name "Test Process UMITOOLS_EXTRACT"
    script "../main.nf"
    process "UMITOOLS_EXTRACT"
    config "./nextflow.config"
    tag "modules_nfcore"
    tag "modules"
    tag "umitools"
    tag "umitools/extract"

    test("Should run without failures") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [ [ id:'test', single_end:true ], // meta map
                    [ file(params.modules_testdata_base_path + "genomics/sarscov2/illumina/fastq/test_1.fastq.gz", checkIfExists: true) ]
                ]
                """
            }
        }

        then {
            assertAll (
            { assert process.success },
            { assert snapshot(process.out.versions).match("versions") }
            )
        }

    }
}