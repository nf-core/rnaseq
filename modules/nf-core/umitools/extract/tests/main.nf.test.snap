{
    "versions": {
        "content": [
            [
                "versions.yml:md5,568d243174c081a0301e74ed42e59b48"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-16T10:01:33.326046137"
    }
}